### ISSUE TITLE
INSERT INTO table FORMAT CSV successfully but reports error msg

### ISSUE THREAD

**Issue by l1t1** [ClickHouse/ClickHouse#18785]
I run the bin/upload-data.sh from https://github.com/Altinity/tpc-ds

```
#!/bin/bash

for file_name in `ls ../data/*.dat`; do
    table_file=$(echo "${file_name##*/}")
    table_name=$(echo "${table_file%_*}" | tr '[:upper:]' '[:lower:]' | tr -d [0-9]|sed 's/__//g')
    upload_data_sql="INSERT INTO $table_name FORMAT CSV"

    echo "$upload_data_sql <-- $(du -h $file_name)"

    cat $file_name | clickhouse client --format_csv_delimiter="|" --max_partitions_per_insert_block=100 --database="tpcds" --query="$upload_data_sql"
    rm $file_name

    sleep 5
done
```
reports
INSERT INTO customer_address FORMAT CSV <-- 5.3M	../data/customer_address_1_64_0001.dat
Cannot set alternative signal stack for thread, errno: 12, strerror: Cannot allocate memory
Cannot set alternative signal stack for thread, errno: 12, strerror: Cannot allocate memory

**Comment by l1t1** [comment]
and if the dat file is bigger, the error msg repeats more times

INSERT INTO store FORMAT CSV <-- 4.0K	../data/store_1_64_0001.dat
Cannot set alternative signal stack for thread, errno: 12, strerror: Cannot allocate memory
Cannot set alternative signal stack for thread, errno: 12, strerror: Cannot allocate memory
INSERT INTO store_returns FORMAT CSV <-- 32M	../data/store_returns_1_64_0001.dat
Cannot set alternative signal stack for thread, errno: 12, strerror: Cannot allocate memory
Cannot set alternative signal stack for thread, errno: 12, strerror: Cannot allocate memory
Cannot set alternative signal stack for thread, errno: 12, strerror: Cannot allocate memory
Cannot set alternative signal stack for thread, errno: 12, strerror: Cannot allocate memory
Cannot set alternative signal stack for thread, errno: 12, strerror: Cannot allocate memory
INSERT INTO store_sales FORMAT CSV <-- 371M	../data/store_sales_1_64_0001.dat
Cannot set alternative signal stack for thread, errno: 12, strerror: Cannot allocate memory
Cannot set alternative signal stack for thread, errno: 12, strerror: Cannot allocate memory
Cannot set alternative signal stack for thread, errno: 12, strerror: Cannot allocate memory
Cannot set alternative signal stack for thread, errno: 12, strerror: Cannot allocate memory
Cannot set alternative signal stack for thread, errno: 12, strerror: Cannot allocate memory
Cannot set alternative signal stack for thread, errno: 12, strerror: Cannot allocate memory
Cannot set alternative signal stack for thread, errno: 12, strerror: Cannot allocate memory
Cannot set alternative signal stack for thread, errno: 12, strerror: Cannot allocate memory
Cannot set alternative signal stack for thread, errno: 12, strerror: Cannot allocate memory
Cannot set alternative signal stack for thread, errno: 12, strerror: Cannot allocate memory

**Comment by l1t1** [comment]
my version https://builds.clickhouse.tech/master/aarch64/clickhouse
and after running `strip clickhouse`

**Comment by l1t1** [comment]
may the set command raise the errors?

```
clickhouse client -mn < query_13.sql
Cannot set alternative signal stack for thread, errno: 12, strerror: Cannot allocate memory
Cannot set alternative signal stack for thread, errno: 12, strerror: Cannot allocate memory
```

```sql
USE tpcds; SET partial_merge_join_optimizations = 1, max_bytes_before_external_group_by = 5000000000, max_bytes_before_external_sort = 5000000000;
select avg(ss_quantity)
       ,avg(ss_ext_sales_price)
       ,avg(ss_ext_wholesale_cost)
       ,sum(ss_ext_wholesale_cost)
 from store_sales
     ,store
     ,customer_demographics
     ,household_demographics
     ,customer_address
     ,date_dim
 where s_store_sk = ss_store_sk
 and  ss_sold_date_sk = d_date_sk and d_year = 2001
 and((ss_hdemo_sk=hd_demo_sk
  and cd_demo_sk = ss_cdemo_sk
  and cd_marital_status = 'D'
  and cd_education_status = '2 yr Degree'
  and ss_sales_price between 100.00 and 150.00
  and hd_dep_count = 3   
     )or
     (ss_hdemo_sk=hd_demo_sk
  and cd_demo_sk = ss_cdemo_sk
  and cd_marital_status = 'S'
  and cd_education_status = 'Secondary'
  and ss_sales_price between 50.00 and 100.00   
  and hd_dep_count = 1
     ) or 
     (ss_hdemo_sk=hd_demo_sk
  and cd_demo_sk = ss_cdemo_sk
  and cd_marital_status = 'W'
  and cd_education_status = 'Advanced Degree'
  and ss_sales_price between 150.00 and 200.00 
  and hd_dep_count = 1  
     ))
 and((ss_addr_sk = ca_address_sk
  and ca_country = 'United States'
  and ca_state in ('CO', 'IL', 'MN')
  and ss_net_profit between 100 and 200  
     ) or
     (ss_addr_sk = ca_address_sk
  and ca_country = 'United States'
  and ca_state in ('OH', 'MT', 'NM')
  and ss_net_profit between 150 and 300  
     ) or
     (ss_addr_sk = ca_address_sk
  and ca_country = 'United States'
  and ca_state in ('TX', 'MO', 'MI')
  and ss_net_profit between 50 and 250  
     ))
;
```

**Comment by azat** [comment]
I guess that you reached some limits:
- either total memory limit
- or, and I think that is more likely, maximum number of vmas (`vm.max_map_count`, default is 65k), you can verify this with smth like `wc -l /proc/$(pgrep clickhouse-serv)/maps`, you can attach `/proc/pid/maps` or/and `/proc/pid/smaps` for further analysis

Also note that since #16346 (which is included into your version) number of used vmas will be increased (+1 for each thread, and default number of threads is 10000).

**Comment by l1t1** [comment]
```
:~/tpc-ds-maste/queries$ pgrep clickhouse-serv
:~/tpc-ds-maste/queries$ pgrep clickhouse
4565
:~/tpc-ds-maste/queries$ wc -l /proc/$(pgrep clickhouse)/maps
152 /proc/4565/maps

:~/tpc-ds-maste/queries$ head /proc/4565/maps
00200000-06a2a000 r--p 00000000 08:02 548284                             /home/a/ch/clickhouse
06a39000-0cbf1000 r-xp 06829000 08:02 548284                             /home/a/ch/clickhouse
0cc01000-0cc04000 r--p 0c9e1000 08:02 548284                             /home/a/ch/clickhouse
0cc13000-0cc2b000 rw-p 0c9e3000 08:02 548284                             /home/a/ch/clickhouse
0cc2b000-0cdde000 rw-p 00000000 00:00 0 
7c20000000-7ee0000000 rw-p 00000000 00:00 0 
7ee8604000-7ee8605000 ---p 00000000 00:00 0 
7ee8605000-7ee8e04000 rw-p 00000000 00:00 0 
7ee8e04000-7ee8e05000 ---p 00000000 00:00 0 
7ee8e05000-7ee9604000 rw-p 00000000 00:00 0 

:~/tpc-ds-maste/queries$ head /proc/4565/smaps
00200000-06a2a000 r--p 00000000 08:02 548284                             /home/a/ch/clickhouse
Size:             106664 kB
Rss:                6564 kB
Pss:                6452 kB
Shared_Clean:        224 kB
Shared_Dirty:          0 kB
Private_Clean:      6340 kB
Private_Dirty:         0 kB
Referenced:         6124 kB
Anonymous:             0 kB
```

**Comment by azat** [comment]
@l1t1 thanks for details!

Indeed the problem was not in the mmap limits of memory limits but in the size of alternative stack, should be fixed in #18832. 
You can try to get new binaries from the `ClickHouse special build check` check report page (once it will be ready)

**Comment by alexey-milovidov** [comment]
@azat BTW, How did you know that this is AArch64 build?

**Comment by azat** [comment]
@alexey-milovidov 

>my version https://builds.clickhouse.tech/master/aarch64/clickhouse

**Comment by l1t1** [comment]
> ClickHouse special build check

what is the url of `ClickHouse special build check`?

**Comment by l1t1** [comment]
> @l1t1 thanks for details!
> 
> Indeed the problem was not in the mmap limits of memory limits but in the size of alternative stack, should be fixed in #18832.
> You can try to get new binaries from the `ClickHouse special build check` check report page (once it will be ready)

there is no Arch64 build at https://clickhouse-builds.s3.yandex.net/0/19ad9e7a5161821e3dcfe0b4ea1f7d85209000fc/clickhouse_build_check/report.html

**Comment by l1t1** [comment]
i guess it is https://clickhouse-builds.s3.yandex.net/0/19ad9e7a5161821e3dcfe0b4ea1f7d85209000fc/clickhouse_special_build_check/report.html 
and saw link failed

**Comment by azat** [comment]
>i guess it is https://clickhouse-builds.s3.yandex.net/0/19ad9e7a5161821e3dcfe0b4ea1f7d85209000fc/clickhouse_special_build_check/report.html

Yes

>and saw link failed

Here is the link to the report page with the link to the binary that you can try - https://clickhouse-builds.s3.yandex.net/18832/46c3266c6a9f927ec5f914f42943cec15e65ea35/clickhouse_special_build_check/report.html

As for your link it, some other patches breaks the build, I guess it is #18811, but that PR does not have failed build.

**Comment by azat** [comment]
>As for your link it, some other patches breaks the build, I guess it is #18811, but that PR does not have failed build.

Should be fixed in #18871

**Comment by l1t1** [comment]
it is fiexed in 21.1 test build
